{
  "severity": "HIGH", 
  "created_at": "2009-10-22T16:30:00Z", 
  "description": "The pygresql module 3.8.1 and 4.0 for Python does not properly support the PQescapeStringConn function, which might allow remote attackers to leverage escaping issues involving multibyte character encodings.", 
  "updated_at": "2009-12-19T06:57:00Z", 
  "packages": [
    {
      "vulnerable_version": "1:3.8.1-1etch1", 
      "name": "pygresql", 
      "source": "DSA", 
      "fixed_version": "1:3.8.1-1etch2", 
      "uid": "DSA-1911-1", 
      "fixed_package": {
        "source": "pygresql", 
        "version": "1:3.8.1-1etch2", 
        "name": "python-pygresql", 
        "sha256": "bd4c3714f57f8ee029ceb2f8553a518b83a03c8079cf54f34d5dd17ecb23b20c", 
        "sha1": "d6d1a0128fa3f59d04c47a97b2a6037a061e6a8e", 
        "md5": "ea6b668eab27ad64d2e7b02e4affc727", 
        "arch": "AMD64", 
        "filename": "python-pygresql_3.8.1-1etch2_amd64.deb"
      }, 
      "vulnerable_package": {
        "source": "pygresql", 
        "version": "1:3.8.1-1etch1", 
        "name": "python-pygresql", 
        "sha256": "3582fc00065e6f06a6d2c88b942645f5cc68b9922583c2639e4005ceb7900035", 
        "sha1": "5078075aef4a1da2bf0e0218076850957965abf6", 
        "md5": "03060f0cd5e8ee068dea13e9c85c3b7f", 
        "arch": "AMD64", 
        "filename": "python-pygresql_3.8.1-1etch1_amd64.deb"
      }
    }, 
    {
      "vulnerable_version": "1:3.8.1-1etch1", 
      "name": "pygresql", 
      "source": "DSA", 
      "fixed_version": "1:3.8.1-1etch2", 
      "uid": "DSA-1911-1", 
      "fixed_package": {
        "source": "pygresql", 
        "version": "1:3.8.1-1etch2", 
        "name": "python-pygresql", 
        "sha256": "b3715d4f88d1559530966a85c8950016b76166e311ad23922a2cdc9465b9ae85", 
        "sha1": "e23068816a60251a12bde26f3c57f948d55cd53b", 
        "md5": "eaec4a360b3af5e4c334126cf870f4fc", 
        "arch": "I386", 
        "filename": "python-pygresql_3.8.1-1etch2_i386.deb"
      }, 
      "vulnerable_package": {
        "source": "pygresql", 
        "version": "1:3.8.1-1etch1", 
        "name": "python-pygresql", 
        "sha256": "2eb584d03c4cd0d4dc3136477aadb6ed2526c23d1c148c42676281987f040c4f", 
        "sha1": "a6b1b57a62916667a7cdc51fc4f4778c40cc391b", 
        "md5": "0e9b9938735e0cac3a2a9670f271c030", 
        "arch": "I386", 
        "filename": "python-pygresql_3.8.1-1etch1_i386.deb"
      }
    }, 
    {
      "vulnerable_version": "1:3.8.1-3", 
      "name": "pygresql", 
      "source": "DSA", 
      "fixed_version": "1:3.8.1-3+lenny1", 
      "uid": "DSA-1911-1", 
      "fixed_package": {
        "source": "pygresql", 
        "version": "1:3.8.1-3+lenny1", 
        "name": "python-pygresql-dbg", 
        "sha256": "cd11f544cbe980e61fa2c14c0b2d9a4d10e786fd84e0e4c3c7d2fa5c252521a6", 
        "sha1": "71670c4dec72777d200f564fdbb7e397ee732e56", 
        "md5": "b1d7283270efa735daf705a7825981e7", 
        "arch": "I386", 
        "filename": "python-pygresql-dbg_3.8.1-3+lenny1_i386.deb"
      }, 
      "vulnerable_package": {
        "source": "pygresql", 
        "version": "1:3.8.1-3", 
        "name": "python-pygresql-dbg", 
        "sha256": "d32c505a8d6b37b4012a488aa3e53eaa476eacbbbb01b5e5308831b8a79847fd", 
        "sha1": "30ae6fc620e06252a90aa862a5c4d282c1716a3d", 
        "md5": "be17e01b99af5b2aaf3bf5df5e0f0b62", 
        "arch": "I386", 
        "filename": "python-pygresql-dbg_3.8.1-3_i386.deb"
      }
    }, 
    {
      "vulnerable_version": "1:3.8.1-3", 
      "name": "pygresql", 
      "source": "DSA", 
      "fixed_version": "1:3.8.1-3+lenny1", 
      "uid": "DSA-1911-1", 
      "fixed_package": {
        "source": "pygresql", 
        "version": "1:3.8.1-3+lenny1", 
        "name": "python-pygresql", 
        "sha256": "d7da5a21d60c380da6f4fd3dd3b4b0341faf6b327e0a52d43f8c9a3120af8dcb", 
        "sha1": "af1ce34cb2c6482f3ebfbcf767a7d2d7f41fc28b", 
        "md5": "fe068a81f497f69a6ad19d495e652e2b", 
        "arch": "AMD64", 
        "filename": "python-pygresql_3.8.1-3+lenny1_amd64.deb"
      }, 
      "vulnerable_package": {
        "source": "pygresql", 
        "version": "1:3.8.1-3", 
        "name": "python-pygresql", 
        "sha256": "8bbed4ecf31b56e80adfc3c3699f1f847e1f10249b674a5f52481fd0c3f2b8fc", 
        "sha1": "4ffd7630d6494980a37112524c56f37f3839336d", 
        "md5": "bdf7ecc76da67624c58b32a10d694dd4", 
        "arch": "AMD64", 
        "filename": "python-pygresql_3.8.1-3_amd64.deb"
      }
    }, 
    {
      "vulnerable_version": "1:3.8.1-3", 
      "name": "pygresql", 
      "source": "DSA", 
      "fixed_version": "1:3.8.1-3+lenny1", 
      "uid": "DSA-1911-1", 
      "fixed_package": {
        "source": "pygresql", 
        "version": "1:3.8.1-3+lenny1", 
        "name": "python-pygresql", 
        "sha256": "36449981cc2d135638a37125bd6fcb53e6c279f644a5f0d310a90ad9e60e0360", 
        "sha1": "08a6eb80ab155467ff8462527b50d9a58a1078aa", 
        "md5": "d27c445d00283fe59fb6a54cfaaa4156", 
        "arch": "I386", 
        "filename": "python-pygresql_3.8.1-3+lenny1_i386.deb"
      }, 
      "vulnerable_package": {
        "source": "pygresql", 
        "version": "1:3.8.1-3", 
        "name": "python-pygresql", 
        "sha256": "f78835c6428d0df605d620d037d2e4f1b6adfcc38956b0d767d9fa7b02b28820", 
        "sha1": "4c448cf061fddc30c324f0952e06b4103ff4f842", 
        "md5": "bc153dc1cfc328241453143ec19fc0ec", 
        "arch": "I386", 
        "filename": "python-pygresql_3.8.1-3_i386.deb"
      }
    }, 
    {
      "vulnerable_version": "1:3.8.1-3", 
      "name": "pygresql", 
      "source": "DSA", 
      "fixed_version": "1:3.8.1-3+lenny1", 
      "uid": "DSA-1911-1", 
      "fixed_package": {
        "source": "pygresql", 
        "version": "1:3.8.1-3+lenny1", 
        "name": "python-pygresql-dbg", 
        "sha256": "6610dc5e03a70facdd4fb1ab3aac8cf0b9ae97864ff988103c5574f98f098450", 
        "sha1": "cdc1aca1a78caf69cd4c463e420babd1385affa5", 
        "md5": "128f1c3033bde64f9a11f5e913d9d973", 
        "arch": "AMD64", 
        "filename": "python-pygresql-dbg_3.8.1-3+lenny1_amd64.deb"
      }, 
      "vulnerable_package": {
        "source": "pygresql", 
        "version": "1:3.8.1-3", 
        "name": "python-pygresql-dbg", 
        "sha256": "4919a8e184930f85b62998f1e786e3a92ddd743e87c79070c2dd255a8bb747e5", 
        "sha1": "fe7a6922602e73cf0926786bc6763e3bc3358e69", 
        "md5": "dac78835d1065713964b4665d3539d65", 
        "arch": "AMD64", 
        "filename": "python-pygresql-dbg_3.8.1-3_amd64.deb"
      }
    }
  ], 
  "name": "CVE-2009-2940"
}